sympy/physics/control/tests/test_lti.py

```diff
@@ -3406,28 +3466,56 @@ def test_conversion():
     raises(ValueError, lambda: TransferFunction(b*s**2 + p**2 - a*p + s, b - p**2, s).rewrite(StateSpace))
 
+    # DiscreteStateSpace to DiscreteTransferFunction for SISO
+    DH1 = DiscreteStateSpace(A1, B1, C1, D1, 0.1)
+    DH3 = DiscreteStateSpace(Matrix([[a0, a1], [a2, a3]]),
+                             B = Matrix([[b1], [b2]]), C = Matrix([[c1, c2]]),
+                             sampling_time = T)
+    dtm1 = DH1.rewrite(DiscreteTransferFunction)
+    raises(TypeError, lambda: DH1.rewrite(TransferFunction)) # H3 is not a continuous system
+    dtm2 = (-DH1).rewrite(DiscreteTransferFunction)
+
+    dtf1 = dtm1[0][0]
+    dtf2 = dtm2[0][0]
+
+    assert dtf1 == DiscreteTransferFunction(12*z + 59, z**2 + 6*z + 8, z, 0.1)
+    assert dtf2.num == -dtf1.num
+    assert dtf2.den == dtf1.den
+    assert dtf1.sampling_time == 0.1
+    assert dtf2.sampling_time == 0.1
+
+    # DiscreteStateSpace to DiscreteTransferFunction for MIMO
+    DH2 = DiscreteStateSpace(A2, B2, C2, D2, T)
+    dtm3 = DH2.rewrite(DiscreteTransferFunction)
+
+    # outputs for input i obtained at Index i-1. Consider input 1
+    assert dtm3[0][0] == DiscreteTransferFunction(2.0*z**3 + 1.0*z**2 - 10.5*z + 4.5, 1.0*z**3 + 0.5*z**2 - 6.5*z - 2.5, z, T)
+    assert dtm3[0][1] == DiscreteTransferFunction(2.0*z**3 + 2.0*z**2 - 10.5*z - 3.5, 1.0*z**3 + 0.5*z**2 - 6.5*z - 2.5, z, T)
+    assert dtm3[0][2] == DiscreteTransferFunction(2.0*z**2 + 5.0*z - 0.5, 1.0*z**3 + 0.5*z**2 - 6.5*z - 2.5, z, T)
+    assert DH3.rewrite(DiscreteTransferFunction) == [[DiscreteTransferFunction(-c1*(a1*b2 - a3*b1 + b1*z) - c2*(-a0*b2 + a2*b1 + b2*z),
+                                                              -a0*a3 + a0*z + a1*a2 + a3*z - z**2, z, T)]]
     # DiscreteTransferFunction to DiscreteStateSpace
     dtf1 = DiscreteTransferFunction(z+5, z**3+2*z**2+4*z+3, z, 0.1)
-    # TODO:
-    # DSS = dtf1.rewrite(DiscreteStateSpace)
-    # A = Matrix([[0, 1, 0], [0, 0, 1], [-3, -4, -2]])
-    # B = Matrix([0, 0, 1])
-    # C = Matrix([[5, 1, 0]])
-    # D = Matrix([[0]])
-    # assert DSS == DiscreteStateSpace(A, B, C, D, 0.1)
+    DSS = dtf1.rewrite(DiscreteStateSpace)
+    A = Matrix([[0, 1, 0], [0, 0, 1], [-3, -4, -2]])
+    B = Matrix([0, 0, 1])
+    C = Matrix([[5, 1, 0]])
+    D = Matrix([[0]])
 
-    # TODO: assert DSS.rewrite(DiscreteTransferFunction)[0][0] == dtf1
+    assert DSS == DiscreteStateSpace(A, B, C, D, 0.1)
 
-    # DiscreteTransferFunction cannot be converted to StateSpace
+    assert DSS.rewrite(DiscreteTransferFunction)[0][0] == dtf1
+
+    # DiscreteTransferFunction cannot be converted to DiscreteStateSpace
     raises(TypeError, lambda: dtf1.rewrite(StateSpace))
 
     # discrete-time Transfer function has to be proper
-    # TODO: raises(ValueError, lambda: DiscreteTransferFunction(b2*z**2 + b1*z + b0, z + a0, z).rewrite(DiscreteStateSpace))
+    raises(ValueError, lambda: DiscreteTransferFunction(b2*z**2 + b1*z + b0, z + a0, z).rewrite(DiscreteStateSpace))
```

Comment:
On line 3480 you raise a TypeError. We should be consistent what error is returned.